src/main/java/org/kairosdb/datastore/cassandra/CassandraDatastore.java

```diff
@@ -659,7 +680,16 @@ private static void filterAndAddKeys(String metricName, SetMultimap<String, Stri
         if (useSplitField != null && !"".equals(useSplitField) && useSplit.size() > 0) {
             // logger.warn("using split lookup: name={} fields={}", useSplitField, useSplit);
             bsShift = 2;
-            bs = m_psQueryRowKeySplitIndex.bind();
+
+            //  If any part of the query window falls before the new split index
+            //  start time, use the old split index:
+            if (m_cassandraConfiguration.isUseNewSplitIndex() && startTime > m_cassandraConfiguration.getNewSplitIndexStartTimeMs()) {
```

Comment:
for the first run we should either remove `startTime > m_cassandraConfiguration.getNewSplitIndexStartTimeMs()` part or make value of `newSplitIndexStartTimeMs` very big (infinitely big) or introduce 2 separate feature toggles: `m_cassandraConfiguration.isUseNewSplitIndexRead()` and `m_cassandraConfiguration.isUseNewSplitIndexWrite()`

Otherwise Kairos will start reading from "new" table immediately after `useNewSplitIndex` becomes true. Because `getNewSplitIndexStartTimeMs` is 0.

Reply:
Use-new-split-index is false by default.  The start time is meant to be set after we have deployed and verified the new table is filling properly.  However, you may be gating on this value too and writes could be blocked.  We can set this to max_long or equivalent in the meantime.

Reply:
(Having it set to true/false + 0 in the conf file like this is meant for a binary on/off equivalent for new deployments, while our existing one would be controlled from its manifests.)